webapp/app.py

```diff
@@ -166,3 +178,93 @@ def get_jira_tasks(webpage_id: int):
         return jsonify(tasks), 200
     else:
         return jsonify({"error": "Failed to fetch Jira tasks"}), 500
+
+
+@app.route("/api/remove_webpage", methods=["POST"])
+@validate()
+@login_required
+def remove_webpage(body: RemoveWebpageModel):
+    """
+    Remove a webpage based on its status.
+    This function handles the removal of a webpage from the system.
+    If the webpage is new and not in the codebase, it deletes the webpage and associated
+    reviewer records from the database. If the webpage pre-exists, it creates a Jira task to remove
+    the webpage from the code repository and updates the webpage status to "TO_DELETE".
+    Args:
+        body (RemoveWebpageModel): The model containing the details of the webpage to be removed.
+    Returns:
+        Response: A JSON response indicating the result of the operation.
+            - If the webpage is not found, returns a 404 error with a message.
+            - If the webpage is successfully deleted or a task is created, returns a 201 status with a success message.
+            - If there is an error during deletion, returns a 500 error with a message.
+    """
+    webpage_id = body.webpage_id
+
+    webpage = Webpage.query.filter(Webpage.id == webpage_id).one_or_none()
+    if webpage is None:
+        return jsonify({"error": "webpage not found"}), 404
+    if webpage.status == WebpageStatus.NEW:
+        try:
+            jira_tasks = JiraTask.query.filter_by(webpage_id=webpage_id).all()
+            if jira_tasks:
+                for task in jira_tasks:
+                    if app.config["JIRA"].change_issue_status(
+                        issue_id=task.jira_id,
+                        transition_id=JiraStatusTransitionCodes.REJECTED.value,
+                    ):
+                        JiraTask.query.filter_by(id=task.id).delete()
+
+                db.session.commit()
+            Reviewer.query.filter_by(webpage_id=webpage_id).delete()
+            Webpage.query.filter_by(id=webpage_id).delete()
+            db.session.commit()
+
+        except SQLAlchemyError as e:
+            # Rollback if there's any error
+            db.session.rollback()
+            print(f"Error deleting webpage: {str(e)}")  # log error
```

Comment:
You should use `app.logger` to get the logs in the stdout, it will include other useful info. 
Also, if we're returning a 500 error, we might as well just raise the error itself, as the user won't know the difference

Reply:
Like tell the user about the exact error? But this statement here catches the database error which is mostly an internal thing  not working correctly and I though we should not share such information with user.

Reply:
The error won't bubble up to the user if we're not in debug mode - and we might be able to see the error in our grafana logs

Reply:
Also the logger does a bunch of other stuff, like formatting and timestamps